drain.go

```diff
@@ -279,5 +339,76 @@ func (s *server) serveDrain(w http.ResponseWriter, r *http.Request) {
 
 	parseTimer.UpdateSince(parseStart)
 
+	if len(shadowURLs) > 0 {
+		go amplify(r.Header, buf)
+	}
+	w.WriteHeader(http.StatusNoContent)
+}
+
+func (s *server) serveShadowURLs(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
```

Comment:
Should probably lock here, but this is not really a problem with the scale at which we'll modify and read this value.